src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py

```diff
@@ -286,3 +277,26 @@ def miscs_update_idxs_vals(self, miscs, idxs, vals,
                 if assert_all_vals_used or tid in misc_by_id:
                     misc_by_id[tid]['idxs'][key] = [tid]
                     misc_by_id[tid]['vals'][key] = [val]
+    
+    def update_parameters_queue(self):
+        '''update the parameters queue when it is empty or when one trial result is received
+        '''
+        rval = self.rval
+        trials = rval.trials
+        algorithm = rval.algo
+        new_ids = rval.trials.new_trial_ids(self.max_queue_len)
+        rval.trials.refresh()
+        random_state = rval.rstate.randint(2**31-1)
```

Comment:
does 2**31 will cost time?

Reply:
It doesn't cost time and the time is about 3-5 µs.